---
title: 'Analysis of the KRAS datataset'
author: "Hector Roux de BÃ©zieux"
bibliography: ref.bib
output: 
  rmarkdown::html_document:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{Analysis of the KRAS dataset}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


# Load data

The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of three cancer models under a  KRAS(G12C) inhibition [@Xue2020].

```{r}
libs <- c("here", "dplyr", "tradeSeq", "SingleCellExperiment", "slingshot",
           "condiments", "scater", "RColorBrewer", "pheatmap", "cowplot",
          "tidyr", "ggplot2")
suppressMessages(
  suppressWarnings(sapply(libs, require, character.only = TRUE))
)
rm(libs)
theme_set(theme_classic())
```

```{r, eval = FALSE}
kras <- bioc2021trajectories::import_KRAS()
```

```{r}
data("kras", package = "bioc2021trajectories")
kras
```

# EDA

Reduced dimension coordinates are obtained from the original publication.

```{r}
cols <- c(brewer.pal(5, "Blues")[2:5],
          brewer.pal(3, "Greens")[2:3],
          brewer.pal(3, "Reds")[2:3],
          brewer.pal(3, "Oranges")[2], "Grey")
names(cols) <- c(3, 5, 4, 1, 8, 2, 9, 10, 6, 7)
df <- colData(kras)[, -97] %>% as.data.frame() %>%
  sample_frac(1)
ggplot(df, aes(x = tSNE1, y = tSNE2, col = Batch)) +
  geom_point(size = .7) +
  scale_color_brewer(palette = "Accent") +
  labs(col = "Type")
```

```{r}
ggplot(df, aes(x = tSNE1, y = tSNE2, fill = Cluster)) +
  geom_point(size = 1, alpha = .65, col = "grey70", shape = 21) +
  scale_fill_manual(values = cols) +
  labs(fill = "Cell Clusters")
```

# Imbalance score

```{r}
kras <- imbalance_score(Object = kras, conditions = "Batch", dimred = "TSNE")
df$scores <- kras[, df$cells]$scores$scaled_scores
ggplot(df, aes(x = tSNE1, y = tSNE2, col = scores)) +
  geom_point(size = .7) +
  scale_color_viridis_c(option = "C") +
  labs(col = "Score")
```

# Differential Topology

To estimate the trajectory, we use _slingshot_ [@Street2018a].

## Fit slingshot

```{r}
kras <- slingshot(kras, reducedDim = "TSNE", clusterLabels = kras$Cluster,
                 start.clus = 7, extend = "n", reweight = FALSE, reassign = FALSE)
```

## Topology Test

```{r, eval = FALSE}
set.seed(23)
topologyTest(kras, conditions = "Batch", rep = 50)
```

## Individual curves

```{r}
sdss <- slingshot_conditions(kras, kras$Batch, approx_points = FALSE,
                             extend = "n", reweight = FALSE, reassign = FALSE)
sdss$condition_id <- names(sdss)
sdss$mapping <- matrix(rep(1:3, each = 3), nrow = 3, ncol = 3, byrow = TRUE)
sds <- do.call(merge_sds, sdss)
```

```{r}
df <- df %>% select(cells, tSNE1, tSNE2, Cluster, Batch) %>%
  full_join(
    slingPseudotime(sds) %>%
      as.data.frame() %>%
      mutate(cells = rownames(.))) %>%
  pivot_longer(starts_with("Lineage"), names_to = "Curve", values_to = "pst")
p <- ggplot(df, aes(x = tSNE1, y = tSNE2, col = Batch)) +
  geom_point(size = .7, alpha = .1) +
  scale_color_brewer(palette = "Accent")
lineages <- lapply(sdss[1:3], function(crv) {
  slingCurves(crv, as.df = TRUE)
}) %>%
  bind_rows(.id = "Batch") %>%
    arrange(Order)
p <- p + geom_path(data = lineages, size = 1.5, 
                   mapping = aes(group = interaction(Lineage, Batch)))
position <- data.frame(
  "tSNE1" = c(40, -30, 45),
  "tSNE2" = c(50, -50, -50),
  "Batch" = "H2122A",
  "text" = paste0("Lineage ", 1:3)
)
p + geom_text(data = position, col = "black", aes(label = text), size = 5)
```

```{r}
p <- ggplot(df, aes(x = tSNE1, y = tSNE2, col = Batch)) +
  geom_point(size = .7, alpha = .1) +
  scale_color_brewer(palette = "Accent")

cls <- df %>% group_by(Cluster, Batch) %>%
  dplyr::summarise(tSNE1 = mean(tSNE1), tSNE2 = mean(tSNE2), .groups = NULL)
p <- p +
  geom_point(data = cls, size = 3)
edges <- lapply(slingLineages(sds), function(lin) {
  from <- lin[1:(length(lin) - 1)]
  to <- lin[2:length(lin)]
  return(data.frame("from" = from, "to" = to))
}) %>% bind_rows()
for (batch in unique(kras$Batch)) {
  cl_batch <- cls %>% filter(Batch == batch)
  edges_batch <- left_join(edges, cl_batch, by = c("from" = "Cluster")) %>%
    left_join(cl_batch %>% dplyr::rename("tSNE1_end" = "tSNE1",
                                  "tSNE2_end" = "tSNE2") %>%
                select(-Batch), by = c("to" = "Cluster") )
    p <- p +
     geom_segment(data = edges_batch, aes(xend = tSNE1_end, yend = tSNE2_end),
                  size = 2)
}
p
```

# Differential Progression
## Test

```{r}
progressionTest(sds, conditions = kras$Batch, lineages = TRUE)
```

## Plot

```{r}
ggplot(df, aes(x = pst)) +
  geom_density(alpha = .4, aes(fill = Batch), col = "transparent") +
  geom_density(aes(col = Batch), fill = "transparent", size = 1.5) +
  guides(col = FALSE) +
  scale_fill_brewer(palette = "Accent") +
  scale_color_brewer(palette = "Accent") +
  labs(x = "Pseudotime", fill = "Type") +
  facet_wrap(~ Curve, scales = "free_x")
```

# Differential Differentiation
## Test

```{r}
differentiationTest(sds, conditions = kras$Batch, pairwise = TRUE)
```

## Plot

```{r}
weights <- condiments:::.sling_reassign(sds)
df <- df %>%
  full_join(weights %>% 
              as.data.frame() %>%
              mutate(cells = rownames(.)) %>%
              dplyr::rename("Lineage1" = V1, "Lineage2" = V2, "Lineage3" = V3) %>%
              pivot_longer(starts_with("Lineage"), names_to = "Curve", values_to = "weights")
      )
df_w <- df %>%
  select(-pst) %>%
  group_by(cells) %>%
  mutate(weights = weights / sum(weights)) %>%
  pivot_wider(names_from = "Curve", values_from = "weights")
ggplot(df_w, aes(x = Lineage1, y = Lineage3)) +
  # geom_hex(aes(fill=log(..count..))) +
  geom_hex() +
  scale_fill_viridis_c(direction = -1) +
  facet_wrap(~Batch, scales = "free") +
  geom_abline(slope = -1, intercept = 1, linetype = "dotted") +
  geom_abline(slope = -1, intercept = 2/3, linetype = "dotted") +
  geom_abline(slope = -1, intercept = 1/3, linetype = "dotted") +
  annotate("text", x = .53, y = .53, label = "w3 = 0", angle = -52) +
  annotate("text", x = .62, y = .1, label = "w3 = 1/3", angle = -52) +
  annotate("text", x = .14, y = .14, label = "w3 = 2/3", angle = -52) +
  theme(legend.position = "bottom") +
  labs(x = "Weights for Lineage 1 (w1)", y = "Weights for Lineage 2 (w2)",
       fill = "counts per hexagon")
```

```{r}
df_w <- df %>%
  select(-pst) %>%
  group_by(cells) %>%
  mutate(weights = weights / sum(weights)) %>%
  ungroup() %>%
  group_by(Batch, Curve) %>%
  summarise(weights = mean(weights), .groups = NULL)
ggplot(df_w, aes(x = Curve, fill = Batch, y = weights)) +
  geom_col(position = "dodge") +
  scale_fill_brewer(palette = "Accent") +
  theme(legend.position = c(.7, .7)) +
  labs(x = "", y = "Mean weight")
```

# Session info 

```{r}
sessionInfo()
```

# References
