<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Trajectory inference across conditions: overview • bioc2021trajectories</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Trajectory inference across conditions: overview">
<meta property="og:description" content="bioc2021trajectories">
<meta property="og:image" content="https://hectorrdb.github.io/bioc2021trajectories/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bioc2021trajectories</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.21</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/bioc2021trajectories.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/KRAS.html">Second example: KRAS dataset</a>
    </li>
    <li>
      <a href="../articles/TGFB.html">First example: TGF-beta dataset</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/bioc2021trajectories">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="bioc2021trajectories_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Trajectory inference across conditions: overview</h1>
            
      
      
      <div class="hidden name"><code>bioc2021trajectories.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<div id="instructors-names-and-contact-information" class="section level2">
<h2 class="hasAnchor">
<a href="#instructors-names-and-contact-information" class="anchor"></a>Instructor(s) name(s) and contact information</h2>
<ul>
<li>Hector Roux de Bézieux (<a href="mailto:hector.rouxdebezieux@berkeley.edu" class="email">hector.rouxdebezieux@berkeley.edu</a>)</li>
<li>Kelly Street (<a href="mailto:kstreet@ds.dfci.harvard.edu" class="email">kstreet@ds.dfci.harvard.edu</a>)</li>
<li>Koen Van den Berge (<a href="mailto:koenvdberge@berkeley.edu" class="email">koenvdberge@berkeley.edu</a>)</li>
</ul>
</div>
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>In single-cell RNA-sequencing (scRNA-seq), gene expression is assessed at the level of single cells. In dynamic biological systems, it may not be appropriate to assign cells to discrete groups, but rather a continuum of cell states may be observed, e.g. the differentiation of a stem cell population into mature cell types. This is often represented as a trajectory in a reduced dimension of the scRNA-seq dataset.</p>
<p>Many methods have been suggested for trajectory inference. However, in this setting, it is often unclear how one should handle multiple biological groups or conditions, e.g. constructing and comparing the differentiation trajectory of a wild type versus a knock-out stem cell population.</p>
<p>In this workshop, we will explore methods for comparing multiple conditions in a trajectory inference analysis. We will expand on our recent package <a href="https://bioconductor.org/packages/devel/bioc/html/condiments.html">condiments</a> and present the workflow on two examples.</p>
<ul>
<li><p>We will first present a <em>single-lineage trajectory dataset with two conditions</em> which we call <strong>TGFB</strong>, and conduct a full workflow analysis on that dataset, from integration to differential expression. You can find the compiled version of that vignette <a href="https://hectorrdb.github.io/bioc2021trajectories/articles/TGFB.html">here</a>.</p></li>
<li><p>We will then analyze a more complex dataset, with <em>three lineages and three conditions</em>, which we name <strong>KRAS</strong> and explore how the first few steps of the workflow differ. You can find the compiled version of that vignette <a href="https://hectorrdb.github.io/bioc2021trajectories/articles/KRAS.html">here</a>.</p></li>
</ul>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<p>Software:</p>
<ul>
<li>Basic knowledge of <em>R</em> syntax</li>
<li>Familiarity with single-cell RNA-sequencing</li>
<li>Familiarity with the <code>SingleCellExperiment</code> class</li>
</ul>
<p>Background reading:</p>
<ul>
<li>The textbook “Orchestrating Single-Cell Analysis with Bioconductor” is a great reference for single-cell analysis using Bioconductor packages.</li>
<li>The first dataset used in this workshop was originally published in <a href="https://www.nature.com/articles/s41588-019-0489-5">McFaline-Figueroa, et al.</a> “A pooled single-cell genetic screen identifies regulatory checkpoints in the continuum of the epithelial-to-mesenchymal transition.”</li>
<li>The second dataset used in this workshop in <a href="https://doi.org/10.1038/s41586-019-1884-x">Xue, et al.</a> “Rapid non-uniform adaptation to conformation-specific KRAS(G12C) inhibition”</li>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4772-0">Slingshot paper</a></li>
<li><a href="https://www.nature.com/articles/s41467-020-14766-3">tradeSeq paper</a></li>
<li><a href="https://www.biorxiv.org/content/10.1101/2021.03.09.433671v1">condiments paper</a></li>
</ul>
</div>
<div id="workshop-participation" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h2>
<p>The workshop will start with an introduction to the problem and the first dataset using <a href="">presentation slides</a>. Following this, we will have a <a href="https://hectorrdb.github.io/bioc2021trajectories/articles/TGFB.html">lab session on the first dataset</a>. Time permitting, we will then look at <a href="https://hectorrdb.github.io/bioc2021trajectories/articles/KRAS.html">a second dataset</a>.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level2">
<h2 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h2>
<ul>
<li>The workshop will focus on Bioconductor packages <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html">SingleCellExperiment</a>, <a href="https://bioconductor.org/packages/release/bioc/html/slingshot.html">Slingshot</a>, <a href="https://bioconductor.org/packages/release/bioc/html/tradeSeq.html">tradeSeq</a> and <a href="https://bioconductor.org/packages/devel/bioc/html/condiments.html">condiments</a>.</li>
</ul>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>10m</td>
</tr>
<tr class="even">
<td>TGFB: Data Integration</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>TGFB: Trajectory Inference and Differential Topology</td>
<td>10m</td>
</tr>
<tr class="even">
<td>TGFB: Differential Progression</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>TGFB: Differential Expression</td>
<td>10m</td>
</tr>
<tr class="even">
<td>KRAS: Trajectory Inference and Differential Topology</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>KRAS: Differential Progression and Differentiation</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Wrap-up and Conclusions</td>
<td>5m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<p>Participants will learn how to reason about trajectories in single-cell RNA-seq data and how they may be used for interpretation of complex scRNA-seq datasets. Participants can follow along in the following fashions:</p>
<ul>
<li>Run the code online by going to <a href="http://app.orchestra.cancerdatasci.org/1">orchestra</a> and launching the workshop titled <strong>Trajectory inference across conditions: differential expression and differential progression</strong>
</li>
<li>Clone the <a href="https://github.com/hectorrdb/bioc2021trajectories">repository</a> from github and run the code locally</li>
<li>USe docker and get the container hectorrdb/bioc2021trajectories:latest</li>
</ul>
</div>
<div id="learning-goals" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h2>
<ul>
<li>Reason about dynamic biological systems.</li>
<li>Grasp the complexity of analyzing large scRNA-seq datasets with the goal of extracting relevant biological information.</li>
<li>Understand the concepts of differential progression and differential expression along a trajectory path.</li>
</ul>
</div>
<div id="learning-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Learn how to analyze single-cell RNA-seq data using Bioconductor packages.</li>
<li>Import and explore large scRNA-seq datasets.</li>
<li>Understand the challenges of trajectory inference.</li>
<li>Compose analysis pipeline that allows interpretation of complex scRNA-seq datasets.</li>
<li>Assess the added complexity of handling multiple conditions in these dynamic systems and how it influences the analysis pipeline.</li>
<li>Discuss how the analysis pipeline can incorporate this change and evaluate it.</li>
</ul>
</div>
</div>
<div id="final-notes" class="section level1">
<h1 class="hasAnchor">
<a href="#final-notes" class="anchor"></a>Final notes</h1>
<p>A compiled version of the vignette is available on the <a href="https://hectorrdb.github.io/bioc2021trajectories/articles/bioc2021trajectories.html">workshop website</a>.</p>
<p>If you have questions that you could not ask during the workshop, feel free to open an issue on the github repository <a href="https://github.com/hectorrdb/bioc2021trajectories/issues">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] rprojroot_2.0.2   digest_0.6.27     crayon_1.4.1      R6_2.5.0         
##  [5] magrittr_2.0.1    evaluate_0.14     stringi_1.6.2     rlang_0.4.11     
##  [9] cachem_1.0.5      fs_1.5.0          ragg_1.1.3        rmarkdown_2.9    
## [13] pkgdown_1.6.1     textshaping_0.3.5 desc_1.3.0        tools_4.1.0      
## [17] stringr_1.4.0     yaml_2.2.1        xfun_0.24         fastmap_1.1.0    
## [21] compiler_4.1.0    systemfonts_1.0.2 memoise_2.0.0     htmltools_0.5.1.1
## [25] knitr_1.33</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kelly Street, Koen Van den Berge, Hector Roux de Bezieux.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
