<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of the KRAS datataset • bioc2021trajectories</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of the KRAS datataset">
<meta property="og:description" content="bioc2021trajectories">
<meta property="og:image" content="https://hectorrdb.github.io/bioc2021trajectories/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bioc2021trajectories</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.93</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/bioc2021trajectories.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/KRAS.html">Analysis of the KRAS datataset</a>
    </li>
    <li>
      <a href="../articles/TGFB.html">First example: TGF-Beta dataset</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/bioc2021trajectories">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="KRAS_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of the KRAS datataset</h1>
                        <h4 class="author">Hector Roux de Bézieux</h4>
            
      
      
      <div class="hidden name"><code>KRAS.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of three cancer models under a KRAS(G12C) inhibition <span class="citation">(Xue et al. 2020)</span>. Those types of molecules are currently in early-phase clinical trials and a large group of lung adenocarcinoma seem to be able to at least partially evade those treatments. The authors of the original study therefore look at the impact of KRAS(G12C) inhibitors on three models of tumors and show that cells evolve along three lineages.</p>
<p>While the authors used the three models to validate their results, we can also look for differences between the three types. We therefore have a three-lineages trajectory, with three conditions (i.e. the three models).</p>
<p>We will use this dataset as an example of how to perform the first two steps of the condiments workflow under this setting. + We first check if we can fit a single trajectory, which we call <strong>differential topology</strong>. + We then look for large-scale changes, indicative of <strong>differential progression</strong> and <strong>differential differentiation</strong>. # Load data</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>
<span class="va">opts_chunk</span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  eval <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">libs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"here"</span>, <span class="st">"dplyr"</span>, <span class="st">"tradeSeq"</span>, <span class="st">"SingleCellExperiment"</span>, <span class="st">"slingshot"</span>,
           <span class="st">"condiments"</span>, <span class="st">"scater"</span>, <span class="st">"RColorBrewer"</span>, <span class="st">"pheatmap"</span>, <span class="st">"cowplot"</span>,
          <span class="st">"tidyr"</span>, <span class="st">"ggplot2"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">libs</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">libs</span><span class="op">)</span>
<span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kras</span> <span class="op">&lt;-</span> <span class="fu">bioc2021trajectories</span><span class="fu">::</span><span class="fu"><a href="../reference/import_KRAS.html">import_KRAS</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">kras</span> <span class="op">&lt;-</span> <span class="va">kras</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span></code></pre></div>
<p>We rely on normalization conducted in the original paper, and will used the reduced dimension coordinates that are available. We provide a function to access both the raw counts, the reduced dimension coordinates and other cell-level info. Since we will not be performing differential expression, we do not need to retain the genes and we only keep a lightweight version of the object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"kras"</span>, package <span class="op">=</span> <span class="st">"bioc2021trajectories"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kras</span></code></pre></div>
</div>
<div id="eda" class="section level1">
<h1 class="hasAnchor">
<a href="#eda" class="anchor"></a>EDA</h1>
<p>We can visualize all the single cells in a shared reduced dimensional space, according to the tumor model or the clusters from the original publication.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">5</span>, <span class="st">"Blues"</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>,
          <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"Greens"</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>,
          <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"Reds"</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>,
          <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"Oranges"</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">"Grey"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">5</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">8</span>, <span class="fl">2</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">6</span>, <span class="fl">7</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">kras</span><span class="op">)</span><span class="op">[</span>, <span class="op">-</span><span class="fl">97</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">sample_frac</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span>, col <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Type"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span>, fill <span class="op">=</span> <span class="va">Cluster</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">.65</span>, col <span class="op">=</span> <span class="st">"grey70"</span>, shape <span class="op">=</span> <span class="fl">21</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"Cell Clusters"</span><span class="op">)</span></code></pre></div>
</div>
<div id="differential-topology" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-topology" class="anchor"></a>Differential Topology</h1>
<p>The first question we ask is whether we can fit a common trajectory. If the differences between the models are limited, then a common trajectory is possible. If there are large changes, then it is better to fit one trajectory per tumor model. In that latter case, depending on the scale of those changes, it may still be possible to reconcile the trajectories manually.</p>
<p>To assess this, we first use a visual tool called the <em>imbalance_score</em></p>
<div id="imbalance-score" class="section level2">
<h2 class="hasAnchor">
<a href="#imbalance-score" class="anchor"></a>Imbalance score</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kras</span> <span class="op">&lt;-</span> <span class="fu">imbalance_score</span><span class="op">(</span>Object <span class="op">=</span> <span class="va">kras</span>, conditions <span class="op">=</span> <span class="st">"Batch"</span>, dimred <span class="op">=</span> <span class="st">"TSNE"</span><span class="op">)</span>
<span class="va">df</span><span class="op">$</span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">kras</span><span class="op">[</span>, <span class="va">df</span><span class="op">$</span><span class="va">cells</span><span class="op">]</span><span class="op">$</span><span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span>, col <span class="op">=</span> <span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_viridis_c</span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span></code></pre></div>
<p>There are some clear regions of imbalance (especially in cluster 10), although overall the three conditions are well mixed.</p>
<p>To assess whether a common trajectory can be fitted in a more quantitative manner, we will rely on the <em>topologyTest</em>. To do this, we fit a common trajectory (under the null) which will serve as a guide for condition-level trajectories. To estimate the trajectory, we use <em>slingshot</em> <span class="citation">(Street et al. 2018)</span>.</p>
</div>
<div id="fit-slingshot-and-visualize-the-trajectory-skeleton-" class="section level2">
<h2 class="hasAnchor">
<a href="#fit-slingshot-and-visualize-the-trajectory-skeleton-" class="anchor"></a>Fit slingshot and visualize the trajectory skeleton.</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kras</span> <span class="op">&lt;-</span> <span class="fu">slingshot</span><span class="op">(</span><span class="va">kras</span>, reducedDim <span class="op">=</span> <span class="st">"TSNE"</span>, clusterLabels <span class="op">=</span> <span class="va">kras</span><span class="op">$</span><span class="va">Cluster</span>,
                 start.clus <span class="op">=</span> <span class="fl">7</span>, extend <span class="op">=</span> <span class="st">"n"</span>, reweight <span class="op">=</span> <span class="cn">FALSE</span>, reassign <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mst</span> <span class="op">&lt;-</span> <span class="fu">slingMST</span><span class="op">(</span><span class="va">kras</span>, as.df <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">.65</span>, col <span class="op">=</span> <span class="st">"grey70"</span>, shape <span class="op">=</span> <span class="fl">21</span>,
             <span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Cluster</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"Cell Clusters"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span>, data <span class="op">=</span> <span class="va">mst</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_path</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span>, data <span class="op">=</span> <span class="va">mst</span>, <span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="va">Lineage</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The skeleton of the trajectory is this tree structure, that tracks changes, consistently with the original publication. The <em>topologyTest</em> utilizes this tree structure when fitting curves to the cells of each conditions, as well as random sets of the data, and compare whether the two distributions are similar.</p>
</div>
<div id="topology-test" class="section level2">
<h2 class="hasAnchor">
<a href="#topology-test" class="anchor"></a>Topology Test</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span>
<span class="fu">topologyTest</span><span class="op">(</span><span class="va">kras</span>, conditions <span class="op">=</span> <span class="st">"Batch"</span>, rep <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></code></pre></div>
<p>Here, we clearly reject the <em>topologyTest</em>: we should fit a separate trajectory per condition.</p>
</div>
<div id="individual-curves" class="section level2">
<h2 class="hasAnchor">
<a href="#individual-curves" class="anchor"></a>Individual curves</h2>
<div id="fit" class="section level3">
<h3 class="hasAnchor">
<a href="#fit" class="anchor"></a>Fit</h3>
<p>We therefore fit one trajectory per tumor model, or batch, using the <em>slingshot_conditions</em> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sdss</span> <span class="op">&lt;-</span> <span class="fu">slingshot_conditions</span><span class="op">(</span><span class="va">kras</span>, <span class="va">kras</span><span class="op">$</span><span class="va">Batch</span>, approx_points <span class="op">=</span> <span class="cn">FALSE</span>,
                             extend <span class="op">=</span> <span class="st">"n"</span>, reweight <span class="op">=</span> <span class="cn">FALSE</span>, reassign <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-skeletons" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-skeletons" class="anchor"></a>Plot skeletons</h3>
<p>We can plot the skeleton of each trajectory. They are clearly very similar and we can visually map the trajectories: the lineages map well onto one another.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">sdss</span>, <span class="va">slingMST</span>, as.df <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">bind_rows</span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="va">Batch</span><span class="op">)</span>

<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span>, col <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span>, alpha <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="va">msts</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_path</span><span class="op">(</span>data <span class="op">=</span> <span class="va">msts</span>, <span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">Lineage</span>, <span class="va">Batch</span><span class="op">)</span><span class="op">)</span>,
            size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-curves" class="anchor"></a>Plot curves</h3>
<p>Similarly with the curves, we see that the three lineages of each trajectory map to one another across conditions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lineages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">sdss</span>, <span class="va">slingCurves</span>, as.df <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">bind_rows</span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="va">Order</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">position</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  <span class="st">"tSNE1"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">40</span>, <span class="op">-</span><span class="fl">30</span>, <span class="fl">45</span><span class="op">)</span>,
  <span class="st">"tSNE2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="op">-</span><span class="fl">50</span>, <span class="op">-</span><span class="fl">50</span><span class="op">)</span>,
  <span class="st">"Batch"</span> <span class="op">=</span> <span class="st">"H2122A"</span>,
  <span class="st">"text"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Lineage "</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE1</span>, y <span class="op">=</span> <span class="va">tSNE2</span>, col <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span>, alpha <span class="op">=</span> <span class="fl">.2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_path</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lineages</span>, size <span class="op">=</span> <span class="fl">1.5</span>, <span class="fu">aes</span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">Lineage</span>, <span class="va">Batch</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">position</span>, col <span class="op">=</span> <span class="st">"black"</span>, <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">text</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="manual-mapping" class="section level2">
<h2 class="hasAnchor">
<a href="#manual-mapping" class="anchor"></a>Manual mapping</h2>
<p>Although we have fitted a different trajectory per condition, we can therefore proceed to step 2 and look at differential progression and differentiation, by manually mapping the trajectories.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mapping</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">3</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">mapping</span>
<span class="va">sds</span> <span class="op">&lt;-</span> <span class="fu">merge_sds</span><span class="op">(</span><span class="va">sdss</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">sdss</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="va">sdss</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, 
                 condition_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sdss</span><span class="op">)</span>, mapping <span class="op">=</span> <span class="va">mapping</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="differential-progression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential Progression</h1>
<p>Similarly to the TGFB dataset, we can look at large-scale differences between conditions. Specifically, we want to see if the pseudotime distributions are similar across conditions.</p>
<div id="plot" class="section level2">
<h2 class="hasAnchor">
<a href="#plot" class="anchor"></a>Plot</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span>
  <span class="va">df</span> <span class="op">%&gt;%</span> 
    <span class="fu">select</span><span class="op">(</span><span class="va">cells</span>, <span class="va">tSNE1</span>, <span class="va">tSNE2</span>, <span class="va">Cluster</span>, <span class="va">Batch</span><span class="op">)</span>,
  <span class="fu">slingPseudotime</span><span class="op">(</span><span class="va">sds</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate</span><span class="op">(</span>cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"Lineage"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"Curve"</span>, values_to <span class="op">=</span> <span class="st">"pst"</span><span class="op">)</span>

<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">pst</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_density</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.4</span>, <span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"transparent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_density</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>col <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span>, fill <span class="op">=</span> <span class="st">"transparent"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>col <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Pseudotime"</span>, fill <span class="op">=</span> <span class="st">"Type"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">Curve</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span></code></pre></div>
<p>Visually, we can see three phenomenons:</p>
<ul>
<li>In lineage 1, there is a cluster of cells from H2122A around pseudotime 100, and more cells at the end of the lineage for condition H358A.</li>
<li>In lineage 2, there are many more cells from H2122A for earlier pseudotime values and far fewer for later time points.</li>
<li>In lineage 3, there are many more cells from SW1573A for earlier pseudotime values and far fewer for later time points.</li>
</ul>
</div>
<div id="test" class="section level2">
<h2 class="hasAnchor">
<a href="#test" class="anchor"></a>Test</h2>
<p>We can also test this in a more principled manner using the <em>progressionTest</em>. Since we have three conditions, we perform a classifier test (see <span class="citation">(Lopez-Paz and Oquab 2016)</span>). By setting <code>lineages=TRUE</code>, we perform the test for individual lineages and for the full trajectory.</p>
<p>The test statistic reflects the accuracy of the classifier on a balanced test data, so under the null, the expected value is <span class="math inline">\(1/3\)</span>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">progressionTest</span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">kras</span><span class="op">$</span><span class="va">Batch</span>, lineages <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="differential-differentiation" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-differentiation" class="anchor"></a>Differential Differentiation</h1>
<div id="test-1" class="section level2">
<h2 class="hasAnchor">
<a href="#test-1" class="anchor"></a>Test</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">differentiationTest</span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">kras</span><span class="op">$</span><span class="va">Batch</span>, pairwise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-1" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-1" class="anchor"></a>Plot</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu">condiments</span><span class="fu">:::</span><span class="fu">.sling_reassign</span><span class="op">(</span><span class="va">sds</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span>
  <span class="fu">full_join</span><span class="op">(</span><span class="va">weights</span> <span class="op">%&gt;%</span> 
              <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
              <span class="fu">mutate</span><span class="op">(</span>cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"Lineage1"</span> <span class="op">=</span> <span class="va">V1</span>, <span class="st">"Lineage2"</span> <span class="op">=</span> <span class="va">V2</span>, <span class="st">"Lineage3"</span> <span class="op">=</span> <span class="va">V3</span><span class="op">)</span> <span class="op">%&gt;%</span>
              <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"Lineage"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"Curve"</span>, values_to <span class="op">=</span> <span class="st">"weights"</span><span class="op">)</span>
      <span class="op">)</span>
<span class="va">df_w</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pst</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">group_by</span><span class="op">(</span><span class="va">cells</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>weights <span class="op">=</span> <span class="va">weights</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"Curve"</span>, values_from <span class="op">=</span> <span class="st">"weights"</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df_w</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Lineage1</span>, y <span class="op">=</span> <span class="va">Lineage3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="co"># geom_hex(aes(fill=log(..count..))) +</span>
  <span class="fu">geom_hex</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_viridis_c</span><span class="op">(</span>direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">Batch</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">1</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">2</span><span class="op">/</span><span class="fl">3</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">3</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">.53</span>, y <span class="op">=</span> <span class="fl">.53</span>, label <span class="op">=</span> <span class="st">"w3 = 0"</span>, angle <span class="op">=</span> <span class="op">-</span><span class="fl">52</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">.62</span>, y <span class="op">=</span> <span class="fl">.1</span>, label <span class="op">=</span> <span class="st">"w3 = 1/3"</span>, angle <span class="op">=</span> <span class="op">-</span><span class="fl">52</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">.14</span>, y <span class="op">=</span> <span class="fl">.14</span>, label <span class="op">=</span> <span class="st">"w3 = 2/3"</span>, angle <span class="op">=</span> <span class="op">-</span><span class="fl">52</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Weights for Lineage 1 (w1)"</span>, y <span class="op">=</span> <span class="st">"Weights for Lineage 2 (w2)"</span>,
       fill <span class="op">=</span> <span class="st">"counts per hexagon"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df_w</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pst</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">group_by</span><span class="op">(</span><span class="va">cells</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>weights <span class="op">=</span> <span class="va">weights</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">group_by</span><span class="op">(</span><span class="va">Batch</span>, <span class="va">Curve</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">summarise</span><span class="op">(</span>weights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df_w</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Curve</span>, fill <span class="op">=</span> <span class="va">Batch</span>, y <span class="op">=</span> <span class="va">weights</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.7</span>, <span class="fl">.7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">"Mean weight"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Lopez-Paz2016" class="csl-entry">
Lopez-Paz, David, and Maxime Oquab. 2016. <span>“<span>Revisiting Classifier Two-Sample Tests</span>.”</span> <em>Arxiv</em>, October, 1–15. <a href="http://arxiv.org/abs/1610.06545">http://arxiv.org/abs/1610.06545</a>.
</div>
<div id="ref-Street2018a" class="csl-entry">
Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. <span>“<span class="nocase">Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics</span>.”</span> <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.
</div>
<div id="ref-Xue2020" class="csl-entry">
Xue, Jenny Y., Yulei Zhao, Jordan Aronowitz, Trang T. Mai, Alberto Vides, Besnik Qeriqi, Dongsung Kim, et al. 2020. <span>“<span class="nocase">Rapid non-uniform adaptation to conformation-specific KRAS(G12C) inhibition</span>.”</span> <em>Nature</em> 577 (7790): 421–25. <a href="https://doi.org/10.1038/s41586-019-1884-x">https://doi.org/10.1038/s41586-019-1884-x</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kelly Street, Koen Van den Berge, Hector Roux de Bezieux.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
